### PR TITLE
NxTextLink focus state RSC-648

### PR DESCRIPTION
**Pull request by glenhunter**
https://issues.sonatype.org/browse/RSC-648

As discussed in Slack with @alexalyons126 the typical `:focus` glow was removed, but we retained the blue border.

### CODE REVIEW

File: lib/src/base-styles/_nx-globals.scss [hunk 1]
```diff
@@ -15,8 +15,11 @@ $-nx-scrollable-default-height: 400px;
 .nx-text-link {
   @include nx-container-helpers.container-horizontal;
 
+  border: 1px solid transparent;
   color: var(--nx-color-link);
   cursor: pointer;
+  margin: 0 calc(0px - #{var(--nx-spacing-base)});
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-globals.scss:21]: Why is there interpolation syntax here?

**Comment by rpokorny** [comment]
The glow needs to be removed in kaleido

File: lib/src/base-styles/_nx-globals.scss [hunk 2]
```diff
@@ -15,8 +15,11 @@ $-nx-scrollable-default-height: 400px;
...
+  padding: 0 var(--nx-spacing-base);
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-globals.scss:22]: You're margin adjustment should cancel out not only the padding but the border too.  So it's off by one.  Looking at the mockup, the intent appears to be that it be 4px from the content to the _outside_ of the border, so this padding actually needs to be decreased by 1.

**Comment by rpokorny** [comment]
Would `outline` be cleaner?

**Comment by rpokorny** [comment]
Missing border radius

**Comment by glenhunter** [comment]
> Would `outline` be cleaner?

Well..... 
Screen Shot 2021-09-28 at 4 26 20 PM

File: lib/src/base-styles/_nx-globals.scss (hunk 1 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-globals.scss]: > Why is there interpolation syntax here?

Weird muscle memory I think: https://github.com/sonatype/sonatype-react-shared-components/pull/502/commits/f306a7ab262b132447388637a981726832f6b0ca

File: lib/src/base-styles/_nx-globals.scss (hunk 2 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-globals.scss]: > You're margin adjustment should cancel out not only the padding but the border too. So it's off by one. Looking at the mockup, the intent appears to be that it be 4px from the content to the _outside_ of the border, so this padding actually needs to be decreased by 1.

Fixed here and confirmed by measuring screenshots: https://github.com/sonatype/sonatype-react-shared-components/pull/502/commits/f306a7ab262b132447388637a981726832f6b0ca

**Comment by glenhunter** [comment]
> Missing border radius

https://github.com/sonatype/sonatype-react-shared-components/pull/502/commits/f306a7ab262b132447388637a981726832f6b0ca

**Comment by glenhunter** [comment]
> Would `outline` be cleaner?

Too many cross browser issues

File: lib/src/base-styles/_nx-globals.scss (hunk 2 above)
Comment (rpokorny) [reply lib/src/base-styles/_nx-globals.scss]: > Fixed here and confirmed by measuring screenshots: f306a7a

That has too little margin.  Margin should cancel out padding + border

File: lib/src/base-styles/_nx-globals.scss (hunk 1 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-globals.scss]: > > Fixed here and confirmed by measuring screenshots: [f306a7a](https://github.com/sonatype/sonatype-react-shared-components/commit/f306a7ab262b132447388637a981726832f6b0ca)
> 
> That has too little margin. Margin should cancel out padding + border

In the commit above padding is 4px - 1px = 3px + 1px border = 4px total. The margin is 4px.

File: lib/src/base-styles/_nx-globals.scss (hunk 2 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-globals.scss]: > > Fixed here and confirmed by measuring screenshots: [f306a7a](https://github.com/sonatype/sonatype-react-shared-components/commit/f306a7ab262b132447388637a981726832f6b0ca)
> 
> That has too little margin. Margin should cancel out padding + border

This should fix it: https://github.com/sonatype/sonatype-react-shared-components/pull/502/commits/3d2d12276dd6f1faed2670ba8f494fcb7e139983

**Review by rpokorny — APPROVED** [review]

**Pull request merged**